starky/src/permutation.rs

```diff
@@ -120,6 +118,38 @@ fn compute_permutation_z_poly<F: Field>(
     PolynomialValues::new(partial_products)
 }
 
+/// Computes the reduced polynomial, `\sum beta^i f_i(x) + gamma`, for both the "left" and "right"
+/// sides of a given `PermutationPair`.
+fn permutation_reduced_polys<F: Field>(
+    instance: &PermutationInstance<F>,
+    trace_poly_values: &[PolynomialValues<F>],
+    degree: usize,
+) -> (PolynomialValues<F>, PolynomialValues<F>) {
+    let PermutationInstance { pair, challenge } = instance;
+    let PermutationPair { column_pairs } = pair;
+    let PermutationChallenge { beta, gamma } = challenge;
```

Comment:
nit: I guess it's a matter of preference, but maybe 
```rust
let PermutationInstance {
    pair: PermutationPair { column_pairs },
    challenge: PermutationChallenge { beta, gamma },
} = instance;
```
is slightly less verbose.

Reply:
I agree it seem a bit nicer